// athom-api/index.mjs
import fetch from './src/lib/util';
import './src/lib/types';
import HomeyAPI from './src/lib/HomeyAPI';
import AthomCloudAPI from './src/lib/AthomCloudAPI';
import AthomAppsAPI from './src/lib/AthomAppsAPI';
import AthomCallbackAPI from './src/lib/AthomCallbackAPI';
import AthomWebhooksAPI from './src/lib/AthomWebhooksAPI';
import AthomNotificationsAPI from './src/lib/AthomNotificationsAPI';
import AthomSignalsAPI from './src/lib/AthomSignalsAPI';
import AthomConnectAPI from './src/lib/AthomConnectAPI';
import AthomFirmwareAPI from './src/lib/AthomFirmwareAPI';
import EventEmitter from './src/lib/util/EventEmitter';

if(typeof window === 'object') {
    window.HomeyAPI = HomeyAPI;
    window.AthomCloudAPI = AthomCloudAPI;
    window.AthomAppsAPI = AthomAppsAPI;
    window.AthomCallbackAPI = AthomCallbackAPI;
    window.AthomWebhooksAPI = AthomWebhooksAPI;
    window.AthomNotificationsAPI = AthomNotificationsAPI;
    window.AthomSignalsAPI = AthomSignalsAPI;
    window.AthomConnectAPI = AthomConnectAPI;
    window.AthomFirmwareAPI = AthomFirmwareAPI;
}

export {HomeyAPI, AthomCloudAPI, AthomAppsAPI, AthomCallbackAPI, AthomWebhooksAPI, AthomNotificationsAPI, AthomSignalsAPI, AthomConnectAPI, AthomFirmwareAPI, EventEmitter};
export default {HomeyAPI, AthomCloudAPI, AthomAppsAPI, AthomCallbackAPI, AthomWebhooksAPI, AthomNotificationsAPI, AthomSignalsAPI, AthomConnectAPI, AthomFirmwareAPI, EventEmitter};
